components/hidden_data_box/wikis/ageofempires/hidden_data_box_custom.lua

```diff
@@ -41,4 +51,31 @@ function CustomHiddenDataBox:addCustomVariables(args, queryResult)
 	BasicHiddenDataBox:checkAndAssign('tournament_gamemode', args.gamemode, queryResult.gamemode)
 end
 
+function CustomHiddenDataBox.validateTier(tierString, tierMode)
+	if String.isEmpty(tierString) then
+		return nil, nil
+	end
+	local warning
+	local tierValue = tierString
+	-- tier should be a number defining a tier
+	if tierMode == TIER_MODE_TIERS and not Logic.isNumeric(tierValue) then
+		tierValue = Tier.number[tierValue:lower()] or tierValue
+	end
+	local cleanedTierValue = Tier.text[tierMode][(tierValue):lower()]
+	if not cleanedTierValue then
+		cleanedTierValue = tierString
+		warning = String.interpolate(
+			INVALID_TIER_WARNING,
+			{
+				tierString = tierString,
+				tierMode = tierMode == TIER_MODE_TYPES and 'Tiertype' or 'Tier',
+			}
+		)
+	end
+
+	tierValue = (tierMode == TIER_MODE_TYPES and cleanedTierValue) or tierValue or tierString
```

Comment:
```suggestion
	tierValue = (tierMode == TIER_MODE_TYPES and cleanedTierValue) or tierValue
```
If i am not mistaken, `tierValue` won't ever be nil here, so it won't ever return `tierString`.